spec/on_page_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'spec_helper'
```

Comment:
Check your code for these passing: https://gist.github.com/titusfortner/b23f59af1e7a4ee99ee91045140b2f50
(there's a tricky but significant implementation bug with what you have and you aren't testing the right things to catch it)